openmrs/api/src/main/java/org/projectbuendia/openmrs/api/impl/ProjectBuendiaServiceImpl.java

```diff
@@ -13,23 +13,28 @@
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.openmrs.Encounter;
 import org.openmrs.api.impl.BaseOpenmrsService;
 import org.projectbuendia.openmrs.api.ProjectBuendiaService;
 import org.projectbuendia.openmrs.api.db.ProjectBuendiaDAO;
 
+import javax.annotation.Nullable;
+import java.util.Date;
+import java.util.List;
+
 /** It is a default implementation of {@link ProjectBuendiaService}. */
 public class ProjectBuendiaServiceImpl extends BaseOpenmrsService implements ProjectBuendiaService {
     protected final Log log = LogFactory.getLog(this.getClass());
 
     private ProjectBuendiaDAO dao;
 
-    /** @return the dao */
-    public ProjectBuendiaDAO getDao() {
-        return dao;
+    @Override
+    public void setDAO(ProjectBuendiaDAO dao) {
+        this.dao = dao;
     }
 
-    /** @param dao the dao to set */
-    public void setDao(ProjectBuendiaDAO dao) {
-        this.dao = dao;
+    @Override
+    public List<Encounter> getEncountersModifiedOnOrAfter(@Nullable Date date) {
```

Comment:
It looks like Hibernate has a `ScrollableResults` class that we can use instead. We could make it return an iterator to support pagination further up the stack - that's probably a better design than passing pagination info down the stack - but I'll do this as part of the pagination changes instead of now.